src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs

```diff
@@ -25,6 +27,11 @@ internal static partial class EnumerableHelpers
         /// </returns>
         internal static T[] ToArray<T>(IEnumerable<T> source, out int length)
         {
+            // Copied from Array.MaxLength in System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Array.cs
+            const int ArrayMaxLength = 0X7FFFFFC7;
```

Comment:
IIRC on netfx the value is actually 0X7FEFFFFF (for T other than bytes). Probably doesn't matter, though.